jsettlers.logic/src/jsettlers/logic/movable/strategies/BuildingWorkerStrategy.java

```diff
@@ -393,7 +394,7 @@ private void jobFailed() {
 	}
 
 	private ShortPoint2D getCurrentJobPos() {
-		return building.calculateRealPoint(currentJob.getDx(), currentJob.getDy());
+		return new RelativePoint(currentJob.getDx(), currentJob.getDy()).calculatePoint(building.getPos());
```

Comment:
The JIT should be able to optimize this in such simple cases, so this is OK for PC. The Android JIT seems to have more problems optimizing this and removing the object allocation.

Changes like this are always dangerous in that they might introduce a massive creation of short-lived objects. We should not worry about this too much but this was the case why we use calculateRealPoint.

But you could make the job class extend RelativePoint, that way you save the extra object allocation and code gets cleaner.